### ISSUE TITLE
Wrong: Missing semicolon (semicolon) warning

### ISSUE THREAD

**Issue by DerAlbertCom** [prograhammer/vscode-tslint-vue#15]
When a .vue template is missing the <script type=ts></script> the ts-line-vue has false positive  warnings

![grafik](https://user-images.githubusercontent.com/136992/39431270-77c1f4ba-4c90-11e8-83ea-c563ee7d46aa.png)

expected behavior is no warnings in that case, like here

![grafik](https://user-images.githubusercontent.com/136992/39431384-d8bbb706-4c90-11e8-936d-f5f38dab570f.png)

**Comment by jgalentine007** [comment]
I noticed the same thing when I just spun up a template project with vue-cli 3.0. Which I guess is just a requirement (as listed on the quickstart.)

**Comment by ffxsam** [comment]
I'm experiencing this too, and it's driving me mad. I think I need to remove the extension.

**Comment by justindeng0405** [comment]
I noticed the same thing too. If I disable the extension, I can't find another one to lint Vue SFC.

**Comment by ffxsam** [comment]
Fixed it.

Turns out I had some bad code. Make sure that:

1. all your SFCs are `<script lang="ts">`. I had mixed script blocks, some just plain JS. This was causing issues. And BTW, this is not a bug with this extension, but rather VS Code's parser, it seems.
2. you don't have any type errors anywhere else. Sometimes it seems like there's a domino effect and it wreaks havoc in the project.

**Comment by Nfinished** [comment]
Happens even in an SFC without script tags.

![ex](https://user-images.githubusercontent.com/1719791/43205996-a3a24aa8-8ff2-11e8-851c-0e2a44779e89.png)

**Comment by ffxsam** [comment]
@Nfinished Right. There's nothing wrong with this file, obviously, but it's other files affecting this one. (see my post above yours)

**Comment by Nfinished** [comment]
𝓬𝓻𝓲𝓹𝓮𝓼

**Comment by ffxsam** [comment]
I know, tell me about it. I spent hours trying to figure out what was wrong. I still don't understand why having some components use JS and some use TS is a problem. But it was causing issues in VS Code, as Prettier wouldn't work either.

**Comment by ffxsam** [comment]
I will say this, though: it's not possible to have a .vue file without a `<script>` block. I have to add `<script lang="ts"></script>` just to shut up the "missing semicolon" bug.

@prograhammer Is this something you could address?

**Comment by ffxsam** [comment]
Woops. Actually, all this code was needed:

```vue
<script lang="ts">
import Vue from 'vue';
export default Vue.extend();
</script>
```

Otherwise, `main.ts` was yelling at me because it didn't like `App.vue` being imported.

**Comment by romansp** [comment]
I use this extension together with `eslint-plugin-vue` and `eslint` to get template linting in .vue files. I have [vue/html-self-closing](https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/html-self-closing.md) enabled.

It works fine most of the time, but as soon as you have a component with self-closing `<textarea />` or `<iframe />` (maybe some other tags) it will start reporting `[tslint] Missing semicolon (semicolon)`. Auto-fixing further will break syntax.

Before:
```
<template>
  <textarea />
</template>

<script lang="ts">
import Vue from "vue";
export default Vue.extend({});
</script>
```

After:
```
<template>
  <te"vue"a />
</template>

<;script lang="ts">
import Vue from "vue";
export default Vue.extend({});
</script>
```

After it becomes very annoying I temporarily disable auto-fixing, or execute VSCode's [save without formatting](https://code.visualstudio.com/updates/v1_28#_save-without-formatters).
